[PM-30268] fix(desktop): remove obsolete LockScreen config from Microsoft Store

majiayu000 opened the pull request:
## Summary
- Remove BadgeLogo.png to prevent Bitwarden from appearing as lock screen status app
- The new client no longer supports Windows 10/11 live tiles

Fixes #16746

## Test plan
- Build AppX package after this change
- Verify Bitwarden no longer appears in Windows Settings > Personalization > Lock screen status

majiayu000 committed:
fix(desktop): remove obsolete LockScreen config from Microsoft Store

Remove BadgeLogo.png which caused Bitwarden to appear as a selectable
lock screen status app in Windows Settings. The new client no longer
supports live tiles on Windows 10/11, so this feature is obsolete.

Fixes #16746

Signed-off-by: majiayu000 <1835304752@qq.com>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/bitwarden/clients?pullRequest=18137) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/bitwarden/clients?pullRequest=18137) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/bitwarden/clients?pullRequest=18137) it.</sub>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/bitwarden/clients?pullRequest=18137) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/bitwarden/clients?pullRequest=18137) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/bitwarden/clients?pullRequest=18137) it.</sub>

djsmith85 committed:
Merge branch 'main' into fix/16746-remove-lockscreen-config

djsmith85 commented:
@iinuwa: You recently did some work around the appx builds and also introduced [custom-appx-manifest.xml](http://github.com/bitwarden/clients/blob/main/apps/desktop/custom-appx-manifest.xml#L99) which also lists the BadgeLogo along with `<uap:LockScreen Notification="badgeAndTileText" BadgeLogo="assets\BadgeLogo.png" />`. That line is what is called out as the root issue on the linked ticket. Albeit the custom-appx-manifest.xml currently not in use as that got temporarily reverted, from what I understand.

Any thoughts on this PR? Any experience with the LockScreen tiles?

iinuwa commented:
@djsmith85 I'm not sure of all the implications of removing the Lock Screen status are (though it seems like a good idea to me). This PR would solve that problem correctly. If we move forward with this, the `<uap:LockScreen>` element should also be updated in `custom-appx-manifest.xml` (see below).

> Albeit the custom-appx-manifest.xml currently not in use as that got temporarily reverted, from what I understand.

Correct, that manifest is currently only used locally for cross-compiled builds. We will have to start using the custom manifest when the native passkey plugin is released, so `custom-appx-manifest.xml` should also be updated to remove the lock screen configuration to keep it in sync with what electron-builder produces so that this does not regress.

djsmith85 reviewed (changes requested):
Hi @majiayu000 and thank you for your contribution ❤️ 

Could you please merge the latest from main and update/remove the line in [custom-appx-manifest.xml](https://github.com/bitwarden/clients/blob/main/apps/desktop/custom-appx-manifest.xml#L99).

Once that is done, the resulting build artifact can be tested.

majiayu000 committed:
Merge remote-tracking branch 'upstream/main' into fix/16746-remove-lockscreen-config

majiayu000 committed:
fix(desktop): sync appx manifest lock screen removal

majiayu000 commented:
Updated: merged the latest main and removed the LockScreen configuration from apps/desktop/custom-appx-manifest.xml to keep it in sync with the generated AppX manifest.

djsmith85 committed:
Merge branch 'main' into fix/16746-remove-lockscreen-config
